include/cupla/device/math/Common.hpp

```diff
@@ -39,21 +40,24 @@ namespace detail
 {
     /** Get the concept implementation of the current accelerator
      *
+     * @tparam T_AccOrMathImpl accelerator or math implementation [type alpaka::acc::* or alpaka::math::MathStdLib]
      * @tparam T_Concept alpaka concept
      * @return implementation of the concept
      */
     ALPAKA_NO_HOST_ACC_WARNING
-    template< typename T_Concept >
-    ALPAKA_FN_HOST_ACC  auto getConcept()
+    template< typename T_AccOrMathImpl, typename T_Concept >
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto getConcept()
     {
-        using AccThreadSeqMathConcept = alpaka::concepts::ImplementationBase<
+        using AccMathConcept = alpaka::concepts::ImplementationBase<
             T_Concept,
-            AccThreadSeq
+            T_AccOrMathImpl
         >;
-        using AccMathConcept = alpaka::concepts::ImplementationBase<
+#if CUPLA_DEVICE_PATH == 1
+        using AccThreadSeqMathConcept = alpaka::concepts::ImplementationBase<
```

Comment:
Is this logic right: if we are on the CUDA device compilation path, the check should be performed?

Reply:
Ohh yes this should be enabled for CPU accelerators, there we have the possibility to mix block sequential and block parallel accelerator. Well spotted +1